### DISCUSSION TITLE
Third party scripts

### DISCUSSION THREAD

**Discussion by elmeerr** [remix-run/remix#4303] [Q&A]
Hi, 
I just opening this because I can't really find a good explanation anywhere on the docs of forums. 
I have a pure react version of a website: https://encarts-dev.web.app/ (it is in Portuguese but it does not really matter) and I am using [FontAwesome](https://fontawesome.com/) kits which is injected via script (you can check opening devtools)

The thing is for that website, there is no problem with the icons. It does not matter how many times I refresh, the icons are always there, rendered correctly.

However, I have a new version of the website I created using Remix: https://encartsdev.web.app/ and on this version, the icons are sometimes rendered completely messed up. If you keep refreshing you will see that sometimes it works, sometimes it does not. 

So, I am wondering could be this a Remix issue? that's the only difference between the two versions and I don't really know how to debug it. I even moved the script to the `head` section but the problem remains.

Can anyone help me with that?

Thanks in advance.

**Comment by kiliman** [comment]
Looks like the FontAwesome stylesheets are not getting rendered.

Here's the first icon and not a single one of the FA classes are defined. That's why the size is wrong, because it doesn't define `fa-7x` which determines the size.

Do you have a simple repo to test?

<img width="1038" alt="image" src="https://user-images.githubusercontent.com/47168/193289755-2e0b17fc-c179-4e28-97a6-f4746b846ba2.png">

**Reply by dusty** [reply]
Maybe you need to include the css files in a LinksFunction.  

https://fontawesome.com/docs/web/setup/host-yourself/webfonts#reference-font-awesome-in-your-project

**Reply by elmeerr** [reply]
@dusty should not be the case as this is only necessary if I decide to host the files myself...I am using their "kit" script (pro user stuff) which should give me everything I need (like the old site, first link I provided)

@kiliman yeah...the problem is indeed things from FA is not rendering sometimes and that's basically my problem/question: why it does not get rendered sometimes?

if you keep refreshing (hard reload, clean cache, etc) you will see things works like expected...the issue happens from time to time and I want to understand why.

what I can share: it is just a regular create-remix-app without templates.
root file looks like

```
export default function App() {
  return (
    <html lang="en">
      <head>
        <Meta />
        <Links />
      </head>
      <body>
        <Outlet />
        <ScrollRestoration />
        <script src="https://kit.fontawesome.com/fa-script.js" crossOrigin="anonymous" /> // this is direct from FA website, it works perfectly on the old non-remix site
        <Scripts />
        <LiveReload />
      </body>
    </html>
  );
}

```
  relevant package.json
`````
"dependencies": {
    "@remix-run/node": "^1.6.5",
    "@remix-run/react": "^1.6.5",
    "@remix-run/serve": "^1.6.5",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "devDependencies": {
    "@remix-run/dev": "^1.6.5",
    "@remix-run/eslint-config": "^1.6.5",
    "@types/react": "^17.0.45",
    "@types/react-dom": "^17.0.17",
    "autoprefixer": "^10.4.8",
    "concurrently": "^7.3.0",
    "eslint": "^8.15.0",
    "postcss": "^8.4.14",
    "prettier": "^2.7.1",
    "tailwindcss": "^3.1.7",
    "typescript": "^4.7.4"
  },

**Comment by elmeerr** [comment]
Reply from Font Awesome team:

>Hi Elmer,  Okay, I think I am seeing the issue. It looks like one of the Remix scripts is messing with the Font Awesome stuff. When I slow down  the load time, I can see everything looks good until this file loads:  https://encartsdev.web.app/build/routes/index-COEFTSNK.js  
  I'm not familiar with Remix so I don't know exactly what's happening but I'm guessing that it is rewriting or affecting the Font Awesome javascript which generates the SVGs and the CSS needed to render the icons. If you have any options for avoiding affecting the Font Awesome assets, or having them load last, that might help. Or you might try including the other ways of using Font Awesome, like Web Fonts or hosting yourself.

I was think about that already...when it comes to add scripts to the application it seems to me Remix is very opinionated and this might not always fit good with external dependencies. I am really new to Remix so, is there a way to have more control about how scripts are added to our application? Am I missing anything?
